File: iminling-core/src/main/java/com/iminling/core/annotation/EnableResolve.java
package com.iminling.core.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface EnableResolve {

    ResolveStrategy value() default ResolveStrategy.ALL;

    enum ResolveStrategy{
        NONE,
        ARGUMENTS,
        RETURN_VALUE,
        ALL
    }

}

File: iminling-core/src/main/kotlin/com/iminling/core/annotation/EnableResolve.kt
package com.iminling.core.annotation

import com.iminling.core.constant.ResolveStrategy

/**
 * @author  yslao@outlook.com
 * @since  2021/11/28
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.TYPE, AnnotationTarget.CLASS)
@MustBeDocumented
annotation class EnableResolve(val value: ResolveStrategy = ResolveStrategy.RETURN_VALUE)
